### PR TITLE
remove snapshots testing assertions

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR removes all snapshot testing assertions and fixes most of the tests

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/components/AttributeValue/AttributeValue.styled.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import {Typography} from 'antd';
 import styled, {css} from 'styled-components';
 
-export const ValueText = styled(Typography.Text)<{isCollapsed: boolean}>`
+export const ValueText = styled(Typography.Text)<{$isCollapsed: boolean}>`
   cursor: pointer;
 
-  ${({isCollapsed}) =>
-    isCollapsed
+  ${({$isCollapsed}) =>
```
Comment (xoscar) [inline web/src/components/AttributeValue/AttributeValue.styled.ts:7]: What's the idea around the $? is this causing any warnings or something like that?

File: web/src/components/Diagram/components/DAG.tsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ const Diagram: React.FC<IDiagramProps> = ({trace, selectedSpan, onSelectSpan}):
     );
   }, [trace?.spans]);
 
+  console.log(spanMap);
```
Comment (xoscar) [inline web/src/components/Diagram/components/DAG.tsx:32]: 😬

File: web/src/components/SpanDetail/SpanDetail.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import {capitalize} from 'lodash';
-import {useCallback} from 'react';
+import React, {useCallback} from 'react';
```
Comment (xoscar) [inline web/src/components/SpanDetail/SpanDetail.tsx:2]: I don't think we should need to explicitly import React anymore

File: web/src/components/TestCard/TestCard.tsx [hunk 4]
```diff
@@ -9,8 +9,11 @@ import TestCardActions from './TestCardActions';
 
 interface TTestCardProps {
   test: TTest;
+
```
Comment (xoscar) [inline web/src/components/TestCard/TestCard.tsx:12]: I'd like to see what prettier rules are you using, mine doesn't add the extra whitespace

File: web/src/components/Trace/__tests__/Trace.test.tsx [hunk 5]
```diff
@@ -1,22 +1,101 @@
...
-import TestRunMock from '../../../models/__mocks__/TestRun.mock';
 import {ReduxWrapperProvider} from '../../../redux/ReduxWrapperProvider';
 import Trace from '../Trace';
+import {SemanticGroupNames} from '../../../constants/SemanticGroupNames.constants';
+import {TSpan} from '../../../types/Span.types';
+import {TTestRun} from '../../../types/TestRun.types';
+import {TTest} from '../../../types/Test.types';
+import {HTTP_METHOD} from '../../../constants/Common.constants';
+
+const spanId = '4938u43';
```
Comment (xoscar) [inline web/src/components/Trace/__tests__/Trace.test.tsx:11]: Consider using the mock factories, you can overwrite the different values if you want

File: web/src/gateways/__tests__/Test.gateway.test.ts [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import {endpoints} from '../../redux/apis/TraceTest.api';
 import TestGateway from '../Test.gateway';
 
-const {createTest, getTestById, getTestList, runTest} = endpoints;
+const {createTest, getTestById, getTestList} = endpoints;
 
-jest.mock('../../redux/apis/Test.api', () => {
+jest.mock('../../redux/apis/TraceTest.api', () => {
```
Comment (xoscar) [inline web/src/gateways/__tests__/Test.gateway.test.ts:6]: Good catch!

File: web/src/models/__mocks__/Trace.mock.ts [hunk 7]
```diff
@@ -13,6 +13,7 @@ const TraceMock: IMockFactory<TTrace, TRawTrace> = () => ({
         '1': SpanMock.raw(),
         '2': SpanMock.raw(),
       },
+      spans: [SpanMock.raw(), SpanMock.raw()],
```
Comment (xoscar) [inline web/src/models/__mocks__/Trace.mock.ts:16]: The raw trace doesn't have a spans array, it either comes from the flat or the tree section.

File: web/src/services/Assertion.service.ts [hunk 8]
```diff
@@ -17,10 +17,13 @@ const getValue = (value: string): string => {
 };
 
 const getFilters = (selectors: TSpanSelector[]) =>
-  selectors.map(
-    ({key, operator = CompareOperator.EQUALS, value}) =>
-      `${key}${OperatorService.getOperatorSymbol(operator)}${getValue(value)}`
-  );
+  selectors.map(({key, operator = CompareOperator.EQUALS, value}) => {
```
Comment (xoscar) [inline web/src/services/Assertion.service.ts:20]: I'm changing some of this on my next PR as I'm working on moving the assertions to the backend

**Review by xoscar — CHANGES_REQUESTED** [review]
Hey @cescoferraro GJ! I left some comments and concerns. Let me know what you think

File: web/src/components/AttributeValue/AttributeValue.styled.ts (hunk 1 above)
Comment (cescoferraro) [reply web/src/components/AttributeValue/AttributeValue.styled.ts:7]: https://styled-components.com/docs/api#transient-props

File: web/src/services/Selector.service.ts [hunk 9]
```diff
@@ -46,7 +49,7 @@ const getPseudoSelectorString = (pseudoSelector?: TPseudoSelector): string => {
 const SelectorService = () => ({
   getSelectorString(selectorList: TSpanSelector[], pseudoSelector?: TPseudoSelector): string {
     return selectorList.length
-      ? `span[${getFilters(selectorList).join('  ')}]${getPseudoSelectorString(pseudoSelector)}`
```
Comment (xoscar) [inline web/src/services/Selector.service.ts:49]: We need the double space to make this work, as we are doing a double space split on line 61

File: web/src/services/Selector.service.ts [hunk 10]
```diff
@@ -16,12 +16,15 @@ const getValue = (value: string): string => {
...
-const selectorRegex = /span\[(.*)\]/i;
+const selectorRegex = /span\[(.*)]/i;
 const nthChildNumberRegex = /\((.*)\)/i;
 const operationRegex = /([=]+|contains)/;
 
-const getFilters = (selectors: TSpanSelector[]) =>
-  selectors.map(({key, operator, value}) => `${key} ${operator} ${getValue(value)}`);
+const getFilters = (selectors: TSpanSelector[]) => {
+  return selectors.map(
+    ({key, operator, value}) => `${key}${operator === 'contains' ? ` ${operator} ` : operator}${getValue(value)}`
```
Comment (xoscar) [inline web/src/services/Selector.service.ts:25]: We can always have the spaces between for the operator so I'm not sure that we need to check for `contains` directly.

Comment (xoscar) [reply web/src/services/Selector.service.ts:25]: Let's roll back the changes made around the selectors and we should be good to merge!

**Pull request merged**